packages/yourloops/components/date-pickers/dialog-range-date-picker.tsx

```diff
@@ -0,0 +1,220 @@
+      startDate = endDate;
+      endDate = tmp;
+    }
+    if (maxDate.isBefore(minDate)) {
+      // Swap min/max
+      const tmp = minDate;
+      minDate = maxDate;
+      maxDate = tmp;
+    }
+    if (startDate.isBefore(minDate)) {
+      startDate = minDate;
+    }
+    if (endDate.isAfter(maxDate)) {
+      endDate = maxDate;
+    }
+    return { startDate, endDate, minDate, maxDate };
+  }, [isOpen, props.start, props.end, props.maxDate, props.minDate]);
+
+  const [nextSelection, setNextSelection] = React.useState<"first" | "last">("first");
+  const [selected, setSelected] = React.useState<DatesRange>({ start: startDate, end: endDate });
+  const [selectable, setSelectable] = React.useState<DatesRange | undefined>(undefined);
+
+  React.useEffect(() => {
+    if (isOpen) {
+      // Refresh our states
+      const range: DatesRange = { start: startDate, end: endDate };
+      setNextSelection("first");
+      setSelected(range);
+      if (onSelectedDateChange) {
+        onSelectedDateChange(range.start.format("YYYY-MM-DD"), range.end.format("YYYY-MM-DD"));
+      }
+    }
+  }, [isOpen, startDate, endDate, onSelectedDateChange]);
+
+  const handleCancel = () => {
+    props.onResult();
+  };
+  const handleOK = () => {
+    props.onResult(selected.start.format("YYYY-MM-DD"), selected.end.format("YYYY-MM-DD"));
+  };
+
+  const updateSelectedDate = (date: dayjs.Dayjs): void => {
+    let range: DatesRange;
+    if (nextSelection === "first") {
+      setNextSelection("last");
+      range = { start: date, end: date };
+
+      if (typeof maxSelectableDays === "number" && maxSelectableDays > 0) {
+        // Substract 1 day to the value, or we will be able to select
+        // maxSelectableDays+1 days total
+        setSelectable({
+          start: date.subtract(maxSelectableDays - 1, "days"),
+          end: date.add(maxSelectableDays - 1, "days"),
+        });
+      }
+    } else {
+      setNextSelection("first");
+      if (typeof maxSelectableDays === "number") {
+        // Use minDate/maxDate to do the limits
+        setSelectable(undefined);
+      }
+
+      if (date.isAfter(selected.start)) {
+        range = { start: selected.start, end: date };
+      } else {
+        range = { start: date, end: selected.end };
+      }
+    }
+
+    setSelected(range);
+    if (onSelectedDateChange) {
+      onSelectedDateChange(range.start.format("YYYY-MM-DD"), range.end.format("YYYY-MM-DD"));
+    }
+  };
+
+  const contentClasses = clsx(classes.content, {
+    [classes.contentLandscape]: orientation === "landscape",
+    [classes.contentPortrait]: orientation === "portrait",
+  });
+
+  return (
+    <Dialog id="date-picker-dialog" onClose={handleCancel} open={isOpen} PaperProps={{ className: classes.dialogPaper }}>
+      <DialogContent id="calendar-view" className={contentClasses}>
+        <RangeDatePicker
+          selection={{ mode: "range", selected, selectable, maxSelectableDays } as CalendarSelectionRange}
```

Comment:
Here or in RangeDatePicker, won't really change.
But setting it here will lead to less code.

Reply:
> I agree with Tim, this prop could be directly setted in the RangeDatePicker since this one can't display another mode. I don't see where it lead to less code ?